prevent-broker-scale-down-if-contains-partition .md

```diff
@@ -0,0 +1,63 @@
+
+# Prevent scale down of the brokers if they contain partitions
+
+This proposal is about adding the mechanism for preventing the scale down of the brokers in the cluster if a partition is assigned to it.
+In the future, when we introduce the automatic scale up/down of the Kafka Cluster, this mechanism would help to prevent any data loss when we are scaling down the brokers when they have some partitions assigned to them.
+
+## Current situation
+
+Currently, when the number of Kafka replicas is decreased in the Kafka CR the StatefulSet will shut down the highest numbered brokers and there is no check to see if these brokers contain any topic-partition replicas.
+The [documentation](https://strimzi.io/docs/operators/latest/configuring.html#scaling-clusters-str) recommends that all topic-partition replicas are reassigned before doing this scaling action, which means that if someone tries to scale down without re-assigning the brokers, it can cause data loss.
+
+## Motivation
+
+We should definitely introduce some sort of logic which can detect if the broker which is going to be removed still contains the partitions or not.
+If the partitions are still available on the broker which is going to be scaled, then we should get some warning in the logs which will prompt users to do the reassignment and hopefully reduce the chance of data loss.
+
+## Proposal
+
+This proposal suggest how we can add the check to detect if the broker still contains any partitions and what to do if the broker scale down is not possible.
+
+### Checking if the broker contains partition
+The best way to detect if the broker contains any partitions is to query the topic metadata.
+To do this, we will have to make use of the `AdminClient` class.
+We should probably put the logic for this check inside the `scaleDown` method present in the `KafkaReconciler` class.
```

Comment:
"Probably"? In a proposal you should be clear about exactly what you're proposing. 
```suggestion
We will put the logic for this check inside the `scaleDown` method present in the `KafkaReconciler` class.
```